### ISSUE TITLE
[Proposal] Add integration with Tensorboard

### ISSUE THREAD

**Issue by VladSkripniuk** [optuna/optuna#1060]
## Motivation
Tensorboard provides [HParams dashboard](https://www.tensorflow.org/tensorboard/hyperparameter_tuning_with_hparams) which is useful for identifying most promising sets of hyperparameters. It would be nice add custom module to optuna for writing experiment results to tensorboard.

## Description
We can add a callback, which writes experiment results to tensorboard. Below is an example of tuning number of units in a hidden layer, dropout rate and optimizer for a small network on fashion MNIST.

```
import optuna

import tensorflow as tf
from tensorboard.plugins.hparams import api as hp

fashion_mnist = tf.keras.datasets.fashion_mnist

(x_train, y_train),(x_test, y_test) = fashion_mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

HP_NUM_UNITS = hp.HParam('num_units', hp.Discrete([16, 32]))
HP_DROPOUT = hp.HParam('dropout', hp.RealInterval(0.1, 0.2))
HP_OPTIMIZER = hp.HParam('optimizer', hp.Discrete(['adam', 'sgd']))

METRIC_ACCURACY = 'accuracy'

with tf.summary.create_file_writer('logs/hparam_tuning').as_default():
    hp.hparams_config(
        hparams=[HP_NUM_UNITS, HP_DROPOUT, HP_OPTIMIZER],
        metrics=[hp.Metric(METRIC_ACCURACY, display_name='Accuracy')],
    )

def train_test_model(hparams):
    model = tf.keras.models.Sequential([
        tf.keras.layers.Flatten(),
        tf.keras.layers.Dense(hparams[HP_NUM_UNITS], activation=tf.nn.relu),
        tf.keras.layers.Dropout(hparams[HP_DROPOUT]),
        tf.keras.layers.Dense(10, activation=tf.nn.softmax),
    ])
    model.compile(
        optimizer=hparams[HP_OPTIMIZER],
        loss='sparse_categorical_crossentropy',
        metrics=['accuracy'],
    )

    model.fit(x_train, y_train, epochs=1) # Run with 1 epoch to speed things up for demo purposes
    _, accuracy = model.evaluate(x_test, y_test)
    return accuracy

def run(run_dir, hparams):
    with tf.summary.create_file_writer(run_dir).as_default():
        hp.hparams(hparams)  # record the values used in this trial
        accuracy = train_test_model(hparams)
        tf.summary.scalar(METRIC_ACCURACY, accuracy, step=1)

session_num = 0

def tensorboard_callback(study, trial):
    trial_value = trial.value if trial.value is not None else float("nan")

    num_units = trial.params['NUM_UNITS']
    dropout_rate = trial.params['DROPOUT_RATE']
    optimizer = trial.params['OPTIMIZER']

    hparams = {
        HP_NUM_UNITS: num_units,
        HP_DROPOUT: dropout_rate,
        HP_OPTIMIZER: optimizer,
    }

    global session_num
    run_name = "run-%d" % session_num
    run_dir = 'logs_optuna/hparam_tuning/' + run_name
    #run_dir = 'logs_optuna/hparam_tuning/'

    with tf.summary.create_file_writer(run_dir).as_default():
        hp.hparams(hparams)  # record the values used in this trial
        tf.summary.scalar(METRIC_ACCURACY, trial_value, step=1)
    session_num += 1

def objective(trial):

    num_units = trial.suggest_int('NUM_UNITS', HP_NUM_UNITS.domain.values[0], HP_NUM_UNITS.domain.values[1])
    dropout_rate = trial.suggest_uniform('DROPOUT_RATE', HP_DROPOUT.domain.min_value, HP_DROPOUT.domain.max_value)
    optimizer = trial.suggest_categorical('OPTIMIZER', HP_OPTIMIZER.domain.values)

    hparams = {
        HP_NUM_UNITS: num_units,
        HP_DROPOUT: dropout_rate,
        HP_OPTIMIZER: optimizer,
    }

    accuracy = train_test_model(hparams)
    return accuracy

study = optuna.create_study(direction='maximize')
study.optimize(objective, n_trials=10, timeout=600, callbacks=[tensorboard_callback])

```
Here is the result in Tensorboard HParams dashboard
![image](https://user-images.githubusercontent.com/10694872/77823362-ec7d7000-70fa-11ea-81a3-380c9f55bf35.png)
